documentation/modules/oauth/con-oauth-authentication-flow.adoc

```diff
@@ -1,12 +1,42 @@
 // Module included in the following assemblies:
 //
-// assembly-oauth.adoc
+// assembly-oauth-authentication.adoc
 
 [id='con-oauth-authentication-flow-{context}']
-= OAuth 2.0 authentication mechanism
+= OAuth 2.0 authentication mechanisms
 
-The Kafka _SASL OAUTHBEARER_ mechanism is used to establish authenticated sessions with a Kafka broker.
+You can configure OAuth 2.0 authentication using one of the following authentication mechanisms:
 
-A Kafka client initiates a session with the Kafka broker using the _SASL OAUTHBEARER_ mechanism for credentials exchange, where credentials take the form of an access token.
+* SASL_OAUTHBEARER
+* SASL_PLAIN
 
-Kafka brokers and clients need to be configured to use OAuth 2.0.
+Both mechanisms allow Kafka clients to establish authenticated sessions with Kafka brokers. 
+However, the authentication flow between a client, the authorization server, and the Kafka broker is different for each mechanism.
+
+.SASL_OAUTHBEARER
+
+Support for the SASL_OAUTHBEARER mechanism is built in to many Kafka clients.
+
+The client initiates a session with the Kafka broker using the SASL OAUTHBEARER mechanism for credentials exchange, where credentials take the form of a JWT access token.
+
+Kafka brokers and clients both need to be configured to use OAuth 2.0. 
+SASL_OAUTHBEARER must be enabled in the listener configuration for the Kafka broker.
+
+.SASL_PLAIN
+
+SASL_PLAIN is a simple authentication mechanism that is supported by all Kafka client tools, including developer tools such as kafkacat.
+
+SASL_PLAIN allows client authentication using either of the following methods:
+
+* OAuth access token
+* OAuth `clientId` and `secret` (the client credentials)
+
+The client must enable SASL_PLAIN and set the `username` and `password`. 
+If the client sets `username` to `access-token`, the authorization server will interpret the password as the access token. 
+Otherwise, the authorization server will interpret the user as the `clientId` and the password as the client secret.
+
+Kafka brokers and clients both need to be configured to use OAuth 2.0. SASL_PLAIN must be enabled in the listener configuration for the Kafka broker.
```

Comment:
I think this makes it more clear that this is just a flag in the OAuth configuration. I think we should also have some example here maybe?

```suggestion
Kafka brokers and clients both need to be configured to use OAuth 2.0. SASL_PLAIN must be enabled in the listener OAuth 2.0 authentication configuration for the Kafka broker.
```

Reply:
I can add a short example YAML. "Configuring OAuth 2.0 support for Kafka brokers" already has three examples.